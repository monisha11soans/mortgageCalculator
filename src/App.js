import React from 'react';
import { Calculator } from './components/Calculator';

function App() {
    return (
        <>
        <h3 className="mortgageCalc">Mortgage Payment Calculator</h3>
        <Calculator />
      </>
    );
}
export default App;