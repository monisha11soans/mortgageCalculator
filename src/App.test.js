import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import React from 'react';
import App from './App';

configure({ adapter: new Adapter() });

describe('Mortgage Component', () => {
    const tree = renderer.create(<App />).toJSON();

    it('should render properly', () => {
      expect(tree).toMatchSnapshot();
    });
});