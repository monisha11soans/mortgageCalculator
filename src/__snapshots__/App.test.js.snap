// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mortgage Component should render properly 1`] = `
Array [
  <h3
    className="mortgageCalc"
  >
    Mortgage Payment Calculator
  </h3>,
  <div
    className="App"
  >
    <div
      className="mortgageCard"
    >
      <div
        className="mortageWidth"
      >
        <div
          className="MuiFormControl-root mortgageItem"
          style={
            Object {
              "margin": "8px",
              "width": "350px",
            }
          }
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink={true}
            htmlFor="housePrice"
          >
            Home Price
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            onClick={[Function]}
            size="small"
          >
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-marginDense"
            >
              <p
                className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
              >
                $
              </p>
            </div>
            <input
              aria-invalid={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              disabled={false}
              id="housePrice"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value={500000}
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-2"
                style={
                  Object {
                    "width": 83,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
        <div>
          <div
            className="MuiFormControl-root mortgageItem"
            style={
              Object {
                "margin": "8px",
                "width": "350px",
              }
            }
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="downPayment"
            >
              Down Payment
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
              size="small"
            >
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-marginDense"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
                >
                  $
                </p>
              </div>
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                id="downPayment"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value={100000}
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-2"
                  style={
                    Object {
                      "width": 83,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div>
          <div
            className="MuiFormControl-root mortgageItem"
            style={
              Object {
                "margin": "8px",
                "width": "350px",
              }
            }
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="amountToBorrow"
            >
              Mortgage Amount
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
              size="small"
            >
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-marginDense"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
                >
                  $
                </p>
              </div>
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={true}
                id="amountToBorrow"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value={0}
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-2"
                  style={
                    Object {
                      "width": 83,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div>
          <div
            className="MuiFormControl-root"
            style={
              Object {
                "margin": "8px",
                "width": "195px",
              }
            }
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="outlined-adornment-amount"
            >
              Interest Rate
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
              size="small"
            >
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                id="outlined-adornment-amount"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value={1.8}
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
                >
                  %
                </p>
              </div>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-2"
                  style={
                    Object {
                      "width": 83,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            className="MuiFormControl-root"
            style={
              Object {
                "margin": "8px",
                "width": "140px",
              }
            }
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink={true}
              id="demo-simple-select-outlined-label"
            >
              Loan Term
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-outlined-label demo-simple-select-outlined"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                id="demo-simple-select-outlined"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                25 years
              </div>
              <input
                aria-hidden={true}
                className="MuiSelect-nativeInput"
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={false}
                tabIndex={-1}
                value={25}
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
                >
                  <span>
                    loanTerm
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiPaper-root MuiPaper-elevation3 MuiPaper-rounded"
        style={
          Object {
            "width": "30%",
          }
        }
      >
        <div
          style={
            Object {
              "padding": "0.75rem 1 rem",
              "textAlign": "left",
            }
          }
        >
          <div
            style={
              Object {
                "padding": "16px",
              }
            }
          >
            <p
              className="monthlyPayment"
            >
              $
              0
            </p>
            <small
              className="fontSummary"
            >
              Your estimated monthly payments are:
            </small>
          </div>
          <div
            style={
              Object {
                "padding": "24px",
              }
            }
          >
            <table
              className="monthlyHeader"
            >
              <tr
                className="summaryHeight"
              >
                <td>
                  Down Payment: 
                </td>
                <td
                  className="paddingSummary"
                >
                  $
                  100000
                </td>
              </tr>
              <tr
                className="summaryHeight"
              >
                <td>
                  Mortgage Amount: 
                </td>
                <td
                  className="paddingSummary"
                >
                  $
                  0
                </td>
              </tr>
              <tr
                className="summaryHeight"
              >
                <td>
                  Total Interest Paid: 
                </td>
                <td
                  className="paddingSummary"
                >
                  $
                  0.00
                </td>
              </tr>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <table
    id="mortgageTable"
  >
    <thead>
      <tr>
        <th>
          Year
        </th>
        <th>
          Interest Paid
        </th>
        <th>
          Total Interest Paid
        </th>
        <th>
          Year Principal Repaid
        </th>
        <th>
          Total Principal Repaid
        </th>
        <th>
          Balance
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          1
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          3
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          4
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          5
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          6
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          7
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          8
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          9
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          10
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          11
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          12
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          13
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          14
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          15
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          16
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          17
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          18
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          19
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          20
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          21
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          22
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          23
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          24
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
      <tr>
        <td>
          25
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
        <td>
          $
          0
        </td>
      </tr>
    </tbody>
  </table>,
]
`;
